src/sentry/static/sentry/app/components/button.tsx

```diff
@@ -9,8 +9,37 @@ import ExternalLink from 'app/components/links/externalLink';
 import InlineSvg from 'app/components/inlineSvg';
 import Tooltip from 'app/components/tooltip';
 
-class Button extends React.Component {
-  static propTypes = {
+/**
+ * The button can actually also be an anchor or React router Link (which seems
+ * to be poorly typed as `any`). So this is a bit of a workaround to recieve
+ * the proper html attributes.
+ */
+type ButtonElement = HTMLButtonElement & HTMLAnchorElement & any;
+
+type Props = {
+  priority?: 'default' | 'primary' | 'danger' | 'link' | 'success';
+  size?: 'zero' | 'micro' | 'small' | 'xsmall' | 'large';
+  align?: 'center' | 'left' | 'right';
+  disabled?: boolean;
+  busy?: boolean;
+  to?: string | object;
+  href?: string;
+  icon?: string;
+  title?: string;
+  external?: boolean;
+  borderless?: boolean;
+  label?: string;
+  tooltipProps?: any;
+  onClick?: (e: React.MouseEvent) => void;
+};
+
+export type ButtonProps = Omit<React.HTMLProps<ButtonElement>, keyof Props | 'ref'> &
```

Comment:
Why do you have to omit keyof Props and also merge in Props again?

Reply:
It will try and coerce the html attribute props with the button props and it will be unable to do so. Same reason we omit ref

Reply:
some of the props may exist in `React.HTMLProps<ButtonElement>`, which can clash with props of the same name in `Props` but with potentially different type structure.

```
// type for ref clashes
type Foo = { ref: number; foo: string; } & { ref: string; };

// fix this by using Omit<T, K>
type Foo = Omit<{ ref: number; foo: string; }, 'ref'> & { ref: string; }; 
```